pyscript.core/esm/interpreter/ruby-wasm-wasi.js

```diff
@@ -24,6 +24,18 @@ export default {
         return interpreter;
     },
     registerJSModule,
+    getGlobal(interpreter, name) {
+        try {
+            return this.run(interpreter, name);
```

Comment:
@JeffersGlass I am confused ... this whole MR is about exposing events **only** as methods, and nothing else ... *Ruby* is also explicitly flagged as experimental ... there is no evaluation at all in here, and if you write wrong code, wrong code throws errors ... what am I misreading here?

Reply:
to clarify further, the Ruby solution was a hack around the fact it doesn't expose any way to grab that global method like every other interpreter does ... but `getGlobals` is both an implementation detail and the documentation explicitly says that there is no evaluation for events, so I don't really follow your thinking in there ... hence I respectfully moved forward, as you presenting evaluated events after being against these was surprising but also completely off the point of this MR you asked for before.

Happy to read your answer, if any, but this comment was a bit odd to grasp to me.

Reply:
@JeffersGlass maybe yu're saying Ruby can have problems but then I'd answer "*did I say RegExp are good for simple cases?*" and I let you think about what we decided to remove from previous MR and signed to allow in here ... I won't add a *RegExp* for Ruby, as it's experimental, and it will change (hopefully for better) in the future.

Reply:
I think perhaps I was unclear - my suggestion for a change is much much smaller than what you are proposing, and only in the documentation 😇. All I'm suggesting is adding a parenthetical in the README section on events, like

```
*Experimental runtimes may use code evaluation as a means of extracting references to these functions; this is an implementation detail, and should not be relied on to execute inline code.
```

Just to have one more chance at telling users not to use that functionality if they stumble across it. Because we've seen over the past year of the project that users _do_ end up relying on parts of implementation details that we don't intend, and if we can document them, we have just a slightly better chance of helping them with that.

If that seems like unnecessary cruft in the documentation, feel free to ignore. Maybe I'm getting too deep while the runtime is still 'experimental.'

Reply:
well there is also a question of whether or not you want to support e.g., `py-click=a.b.c` or `py-click=handlers["something"]` or `py-click=functools.partial(a, 77)`, etc etc. Not clear what the consensus is here.

Reply:
> and if we can document them, we have just a slightly better chance of helping them with that.

OK, so this is a documentation issue, you commented on a piece of code for an experimental runtime I've also admittedly struggle to make work for our final decision.

If it's just docs we're after, please file an issue or a PR to fix docs, but to me *Ruby* is second class citizen in here, until experimental (I've noticed there is an EMScripten version I should just try and maybe flip the table around Ruby being experimental, but there's a lot of APIs missing from it, compared to others, so I am not sure I should).

Reply:
> well there is also a question of whether or not you want to support `a.b.c` ...

I am afraid you folks all opted in for current MR so that question was answered by 3 of you telling me: "*we never want evaluation*" and this is what the MR does.

We can revisit this further, but your thoughts were already mentioned by me in the MR that got closed.

I like to move forward in general though, ew can revisit that MR once anyone (users) complain about stuff (which I see coming, but that's not really on me though).

Reply:
P.S. any `.` ot `['thing']` require *RegExp* but I don't like to repeat myself ... please read again the mentioned MR to find answers to all these questions, as I've already explain reasons I preferred the previous MR. I hope this topic won't come up again until somebody has issues, and then I'll just resurrect the MR you decided wasn't worth going forward with, likely with better changes in it, still as "inspiration" for what I think was already the good thing to do.

Reply:
Well it isn't that we never want evaluation. There are two independent questions here:
1. Do we want the value of the `py-click` attribute to be an expression that is evaled every time the button is clicked or do we want it to be a function that is gotten at initialization time and called every time the button is clicked?
2. If we want the value of `py-click` to be a function, how do we decide how to refer to functions? There are several reasonable options, but for example you could either have it be the name of a global or an expression that evaluates to a callable.

So we haven't talked about this second question previously as far as I'm aware.

Reply:
as reminder: https://github.com/pyscript/pyscript/pull/1574#issuecomment-1621932246 and https://github.com/pyscript/pyscript/pull/1574#issuecomment-1622001544

for @hoodmane https://github.com/pyscript/pyscript/pull/1574#issuecomment-1622026925

that sealed the deal, we should move on.

Reply:
> So we haven't talked about this second question previously as far as I'm aware.

we did! if the content of the attribute requires a whole parser (as not every PL uses `.` and Ruby uses `[:thing]` and who knows what else uses what) and if that thing to use as callback is deep into the code namespace, we have an issue: we need evaluation!

since the whole point of Antonio to which you all agreed was that there are interpreters without evaluation ability to consider, that's what this MR does ... no evaluation, no *RegExp* (another point in that PR) it's all user-facing friction when evaluation was desired ... that's the conclusion from that discussion to me, I don't see how I can avoid evaluation or *RegExp* when both are needed for your suggested use case and for multiple interpreters.

Reply:
P.S. the previous R would've solved and allowed all of this, you voted against it though ... maybe a new syntax is needed for your use cases, or maybe the previous MR should've landed already ... it's not on me to decide though, we should decide as a team, but every decision seems to backfire with afterthoughts when I've reasoned about all those afterthoughts already in that MR. Can we move forward now and tackle this as soon as the time is right, which is when users will have an issue with it?

Reply:
I don't think the point before was considering whether evaluation was possible, it can also depend on the interpreter. Anyways I think we clearly need to have a discussion about what we want the events design to look like and also the meta question of what our consensus process should look like.

Reply:
wasn't that the case in the previous MR where everything was just possible? what else do we need to discuss? but also, how can we avoid evaluation for unknown syntax out there without evaluation or without a RegExp for the fast path? These were my point to move forward the other MR, I don't know what else I can add in further discussions if not pointing at that exact MR I've already commented about this topic. If we all feel like this MR was a mistake, I'll rebase and re-push for that MR.

Reply:
> and also the meta question of what our consensus process should look like.

btw, about this, I fully agree, as back and forward with implementation details doesn't benefit anyone in general: not us, not users.

Reply:
Follow up MR here: https://github.com/pyscript/pyscript/pull/1584

  * addressed `a.b.c` use case still avoiding evaluation (not in *Ruby* but direct methods work)
  * ignored the `a['b c']` case as that's YAGNI, problematic, to me something we should not fix or care about
  * removed all get/set/delete *Global* unnecessary operations because ...
  * added `runEvent` utility, which is the cleanest thing I could think about, and it confines eventually interpreter only issues around events with the interpreter, not for every interpreter (see *Ruby* case)
  * we can still revisit this in the future with ease and add ad-hoc features per interpreter
  * it's so clean I've managed to finally make *Ruby* run as a worker with all the `XWorker` features other interpreters have

I hope this will put an end to this debate at least for the next X months and I'd love to have your OK there too ... it's super clean, it works well, I hope we can move forward for long time with it.

Reply:
P.S. I *could* add `(event)` or `( event )` or ` event` logic to make it more Web compact, but I feel like without evaluation that might be more misleading 'cause as soon as an extra or different argument is passed we're back to *need for evaluation* which I am OK removing from core.